src/directives/onLongPress.ts

```diff
@@ -0,0 +1,137 @@
+import {ElementPart, nothing} from 'lit';
+import {AsyncDirective, directive, PartType} from 'lit/async-directive.js';
+import type {
+  DirectiveParameters,
+  DirectiveClass,
+  DirectiveResult,
+  PartInfo
+} from 'lit/async-directive.js';
+
+type LongPressCallback = (event?: PointerEvent) => void;
+
+const DEFAULT_LONG_PRESS_TIMEOUT_MS = 1000;
+
+class LongPressDirective extends AsyncDirective {
+  /** Element of the directive. */
+  #element!: Element;
+
+  /**
+   * Long press timeout.
+   * This timeout initiates when the pointer is pressed on the
+   * element. It calls user's callback unless cancel events occur
+   * before time's out.
+   */
+  #longPressTimeout?: NodeJS.Timeout;
+
+  /** Time before the timeout runs out. */
+  #longPressTimeoutMs?: number | undefined;
+
+  /** User-defined callback for long-press event */
+  #longPressCallback?: LongPressCallback;
+
+  constructor(partInfo: PartInfo) {
+    super(partInfo);
+    if (partInfo.type !== PartType.ELEMENT) {
+      throw new Error(
+        "Can't bind `onLongPress` directive to anything " +
+          'that is not an element.'
+      );
+    }
+    this.#updateElement((partInfo as ElementPart).element);
+  }
+
+  /** @inheritdoc */
+  render(
```

Comment:
similar to how lit's own async directives work, this `render` method doesn't actually need to do anything other than return `noChange`.

as in the assignments should really happen in `update`:

```ts
public render(
  callback: LongPressCallback,
  callbackTimeoutMs: number = DEFAULT_LONG_PRESS_TIMEOUT_MS
): unknown {
  return noChange;
}

public override update(
  part: ElementPart,
  [callback, callbackTimeoutMs]: DirectiveParameters<this>
): void {
  if (part.element !== this.#element) {
    this.#updateElement(part.element);
  }
  this.#longPressCallback = callback;
  this.#longPressTimeoutMs = callbackTimeoutMs;
  return this.render(callback, callbackTimeoutMs);
}
```

you get `noChange` from lit.

note too i added/changed the following i was also going to comment on:

- a type to `callbackTimeoutMs`
- a return type of `unknown`
- `return` the result of `render` in `update`

Reply:
The reason why I wrote this way is because there are some eslint warning that the arguments in `render` are not being used, maybe it's not important, I just don't like having warnings in my code.

Reply:
you just need to prefix them with `_`, i.e. `_callback`, `_callbackTimeoutMs` (only in the render method)